Automattic/wp-calypso
# Follow-up email-only registration test

#### Changes proposed in this Pull Request

This is a follow-up test to the email-only test described and discussed at pbxNRc-144-p2. That flow auto-generates a site name based on the user's email address and is likely leading to a drop in purchase rate for this flow (pbxNRc-144-p2#comment-2770).

This PR changes the generated site name to one based on random words and adds a note to the checkout page. The note informs the user that they can change the site name at any point.

Here's a screenshot of the updated checkout page:

<img width="937" alt="CleanShot 2021-12-12 at 08 04 22@2x" src="https://user-images.githubusercontent.com/35781181/145714142-fff3a46f-425c-4e0b-951b-1c49c3ca1a83.png">

The new Explat test names are:
`registration_email_only_mobile_random_usernames`
`registration_email_only_desktop_random_usernames`

#### Testing instructions



* Patch your sandbox with D71395 (this updates the api that generates the username)
* Sandbox public-api.wordpress.com
* Checkout this branch and start Calypso
* Add the Abacus bookmarklets for both tests listed above.
* Navigate to `[LOCALCALYPSO]/start/new` in an incognito browser
* Repeat the following steps in both mobile and desktop viewports:
* Assign yourself to the treatment variation for the relevant viewport (it may take a minute or two after clicking the bookmarklet for your variation assignment to propagate).
* Go through the control and treatment variations for each viewport.
* Test the flow through a non-onboarding signup flow (such as `/start/personal`)
* Check non-signup checkout flows and confirm that the user isn't assigned to either test, regardless of viewport. (This check is handled in Abacus).



Related to #58404
